pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -468,8 +473,12 @@ async def _make_request(
         if self._result is not None:
             return self._result  # pragma: no cover
 
-        model_settings, model_request_parameters, message_history, _ = await self._prepare_request(ctx)
-        model_response = await ctx.deps.model.request(message_history, model_settings, model_request_parameters)
+        model_settings, model_request_parameters, message_history, run_context = await self._prepare_request(ctx)
+        token = CURRENT_RUN_CONTEXT.set(run_context)
+        try:
+            model_response = await ctx.deps.model.request(message_history, model_settings, model_request_parameters)
+        finally:
+            CURRENT_RUN_CONTEXT.reset(token)
```

Comment:
Can we move this to a contextmanager method in `_run_context.py`, so we can do `with set_current_run_context(run_context):` here and in the method above?